### PR TITLE
Added full support for Thread of Hope

### PR DESCRIPTION
**Pull request by ppoelzl**
Added full support for:
* Thread of Hope
* The new jewel ring sizes
* Intuitive Leap-like functionality jewels of all sizes
Also:
* Separated jewel radius index from jewel radius label
* Disabled allocation of jewel sockets though Intuitive Leap-like effects

### CODE REVIEW

File: Classes/PassiveTreeView.lua [hunk 1]
```diff
@@ -501,23 +517,38 @@ function PassiveTreeViewClass:Draw(build, viewPort, inputEvents)
...
+				end
+			elseif node.alloc then
+				if jewel and jewel.jewelRadiusIndex then
+					local radData = build.data.jewelRadius[jewel.jewelRadiusIndex]
+					local outerSize = radData.outer * scale
+					local innerSize = radData.inner * scale
+					SetDrawColor(radData.col)
+					DrawImage(self.ring, scrX - outerSize, scrY - outerSize, outerSize * 2, outerSize * 2)
+					DrawImage(self.ring, scrX - innerSize, scrY - innerSize, innerSize * 2, innerSize * 2)
+				end
```
Comment (NickRyder) [inline Classes/PassiveTreeView.lua]: perhaps add comments here?

Comment (ppoelzl) [reply Classes/PassiveTreeView.lua]: Good suggestion, I hope now it's clear why we check for the inner size, among other things.

File: Classes/PassiveTreeView.lua [hunk 2]
```diff
@@ -467,10 +467,30 @@ function PassiveTreeViewClass:Draw(build, viewPort, inputEvents)
...
 					elseif hoverNode.type == "Socket" then
 						-- Hover node is a socket, check if this node falls within its radius and color it accordingly
-						for index, data in ipairs(build.data.jewelRadius) do
-							if hoverNode.nodesInRadius[index][node.id] then
-								SetDrawColor(data.col)
-								break
+						local socket, jewel = build.itemsTab:GetSocketAndJewelForNodeID(hoverNode.id)
+						if jewel
+							and jewel.jewelRadiusIndex
+							and data.jewelRadius[jewel.jewelRadiusIndex].label == "Variable" then
```
Comment (NickRyder) [inline Classes/PassiveTreeView.lua]: I'd make this into a separate bool which is aptly named. `isThreadOfHope` perhaps.

File: Classes/PassiveTreeView.lua [hunk 3]
```diff
@@ -501,23 +521,41 @@ function PassiveTreeViewClass:Draw(build, viewPort, inputEvents)
...
-				local size = radData.rad * scale
-				SetDrawColor(radData.col)
-				DrawImage(self.ring, scrX - size, scrY - size, size * 2, size * 2)				
+			if node == hoverNode then
+				for _, radData in ipairs(build.data.jewelRadius) do
+					local outerSize = radData.outer * scale
+					local innerSize = radData.inner * scale
+					if jewel
+						and jewel.jewelRadiusIndex
+						and data.jewelRadius[jewel.jewelRadiusIndex].label == "Variable" then
```
Comment (NickRyder) [inline Classes/PassiveTreeView.lua]: Likewise use a bool here.

File: Classes/PassiveTreeView.lua [hunk 4]
```diff
@@ -501,23 +521,41 @@ function PassiveTreeViewClass:Draw(build, viewPort, inputEvents)
...
+							DrawImage(self.ring, scrX - innerSize, scrY - innerSize, innerSize * 2, innerSize * 2)
+						end
+					else
+						-- Jewel in socket is not Thread of Hope or similar, draw normal jewel radius
+						if innerSize == 0 then
+							SetDrawColor(radData.col)
+							DrawImage(self.ring, scrX - outerSize, scrY - outerSize, outerSize * 2, outerSize * 2)
+							DrawImage(self.ring, scrX - innerSize, scrY - innerSize, innerSize * 2, innerSize * 2)
+						end
+					end
```
Comment (NickRyder) [inline Classes/PassiveTreeView.lua]: I enjoy the formatting where you separate the conditional out first and then have two separate loops to draw within the if/else as in 475-491.

**Comment by ppoelzl** [comment]
All done, thank you for your great suggestions!

**Comment by ppoelzl** [comment]
Closes #126

**Pull request merged**